Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/TvUtil.java

```diff
@@ -0,0 +1,278 @@
+		TvContractCompat.Channels.COLUMN_BROWSABLE
+	};
+	private static final String LEANBACK_PACKAGE = "com.google.android.tvlauncher";
+
+	public static int getNumberOfChannels(Context context)
+	{
+		Cursor cursor =
+			context.getContentResolver()
+				.query(
+					TvContractCompat.Channels.CONTENT_URI,
+					CHANNELS_PROJECTION,
+					null,
+					null,
+					null);
+		return cursor != null ? cursor.getCount() : 0;
+	}
+
+	public static List<Channel> getAllChannels(Context context)
+	{
+		List<Channel> channels = new ArrayList<>();
+		Cursor cursor =
+			context.getContentResolver()
+				.query(
+					TvContractCompat.Channels.CONTENT_URI,
+					CHANNELS_PROJECTION,
+					null,
+					null,
+					null);
+		if (cursor != null && cursor.moveToFirst())
+		{
+			do
+			{
+				channels.add(Channel.fromCursor(cursor));
+			} while (cursor.moveToNext());
+		}
+		return channels;
+	}
+
+	public static Channel getChannelById(Context context, long channelId)
+	{
+		for (Channel channel : getAllChannels(context))
+		{
+			if (channel.getId() == channelId)
+			{
+				return channel;
+			}
+		}
+		return null;
+	}
+
+	/**
+	 * Updates all Leanback homescreen channels
+	 */
+	public static void updateAllChannels(Context context)
+	{
+		List<Channel> channels = getAllChannels(context);
+		for (Channel channel : channels)
+		{
+			context.getContentResolver()
+				.update(
+					TvContractCompat.buildChannelUri(channel.getId()),
+					channel.toContentValues(),
+					null,
+					null);
+		}
+	}
+
+	public static Uri getUriToResource(Context context, @AnyRes int resId)
+		throws Resources.NotFoundException
+	{
+		Resources res = context.getResources();
+		Uri resUri = Uri.parse(ContentResolver.SCHEME_ANDROID_RESOURCE +
+			"://" + res.getResourcePackageName(resId)
+			+ '/' + res.getResourceTypeName(resId)
+			+ '/' + res.getResourceEntryName(resId));
+		return resUri;
+	}
+
+	/**
+	 * Converts a resource into a {@link Bitmap}. If the resource is a vector drawable, it will be
+	 * drawn into a new Bitmap. Otherwise the {@link BitmapFactory} will decode the resource.
+	 */
+	@NonNull
+	public static Bitmap convertToBitmap(Context context, int resourceId)
+	{
+		Drawable drawable = context.getDrawable(resourceId);
+		if (drawable instanceof VectorDrawable)
+		{
+			Bitmap bitmap =
+				Bitmap.createBitmap(
+					drawable.getIntrinsicWidth(),
+					drawable.getIntrinsicHeight(),
+					Bitmap.Config.ARGB_8888);
+			Canvas canvas = new Canvas(bitmap);
+			drawable.setBounds(0, 0, canvas.getWidth(), canvas.getHeight());
+			drawable.draw(canvas);
+			return bitmap;
+		}
+		return BitmapFactory.decodeResource(context.getResources(), resourceId);
+	}
+
+	/**
+	 * Leanback lanucher requires a uri for poster art, so we take the banner vector,
+	 * make a bitmap, save that bitmap, then return the file provider uri.
+	 */
+	public static Uri buildBanner(GameFile game, Context context)
+	{
+		Uri contentUri = null;
+		int[] vector = game.getBanner();
```

Comment:
Check first if the game has a screenshot. GameCube banners are really ugly, I would rather to use the screenshot if it is available, also it will make it more consistent of how Dolphin works.
```

File screenshotFile = new File(URI.create(game.getScreenshotPath()));
		if (screenshotFile.exists()) {
			contentUri = getUriForFile(context, getFilePrivider(context), screenshotFile);
			context.grantUriPermission(LEANBACK_PACKAGE, contentUri, FLAG_GRANT_READ_URI_PERMISSION);
		} else {
		}
```

Reply:
I didn't realize that was the current functionality. I've never actually taken a screenshot in a game before. :) Added.

Also noticed I was creating the banner each time. Added a check to see if it's been done before.